pkg/client/fake/client_test.go

```diff
@@ -1677,6 +1677,52 @@ var _ = Describe("Fake client", func() {
 		Expect(obj.Object["spec"]).To(BeEquivalentTo("original"))
 	})
 
+	It("should not change the status of known unstructured objects that have a status subresource on update", func() {
+		obj := &unstructured.Unstructured{}
+		obj.SetAPIVersion("v1")
+		obj.SetKind("Pod")
+		obj.SetName("pod")
+		err := unstructured.SetNestedField(obj.Object, "Always", "spec", "restartPolicy")
+		Expect(err).NotTo(HaveOccurred())
+		err = unstructured.SetNestedField(obj.Object, "Pending", "status", "phase")
+		Expect(err).NotTo(HaveOccurred())
+		cl := NewClientBuilder().WithStatusSubresource(obj).WithObjects(obj).Build()
+
+		err = unstructured.SetNestedField(obj.Object, "Never", "spec", "restartPolicy")
+		Expect(err).NotTo(HaveOccurred())
+		err = unstructured.SetNestedField(obj.Object, "Running", "status", "phase")
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(cl.Update(context.Background(), obj)).To(Succeed())
+		Expect(cl.Get(context.Background(), client.ObjectKeyFromObject(obj), obj)).To(Succeed())
```

Comment:
Would you actually mind to do the retrieval and verification using `corev1.Pod` instead of `unstructured.Unstructured`? That I believe would simplify the code a bit but more importantly clarify the intention of the testcase. I had to read the title three times to understand it and that is with the context from the PR. Same for the other testcase.

Reply:
Updated the tests. Sorry for the delay, I was traveling last week.